### PR TITLE
Implement fill(mean)

### LINKED ISSUE #3273: [feature request] Implement fill(mean)
Tracking issue for implementation of `fill(mean)`, this is the default interpolation used in some other time series DBs, and pretty important.

### PR DESCRIPTION
**Pull request by pdf**
Fixes #3273

- [ ] CHANGELOG.md updated
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](http://influxdb.com/community/cla.html) (if not already signed)

### CODE REVIEW

File: tsdb/engine.go [hunk 1]
```diff
@@ -489,6 +489,50 @@ func (e *Executor) processFill(results [][]interface{}) [][]interface{} {
...
+					if i-nullCount[j] <= 0 {
+						// can't fill without previous value
+						nullCount[j] = 0
+						continue
+					}
+					step := (int64toFloat64(vals[j]) - int64toFloat64(results[i-nullCount[j]-1][j])) / float64(nullCount[j]+1)
+					current := int64toFloat64(vals[j])
+					var count int
+					if nullFirst {
+						count = i
```
Comment (pdf) [inline tsdb/engine.go]: So, I'm not really certain if extrapolating the data back to the start like this is a good idea, or whether we should just leave the start of the series with nulls.

**Comment by otoolep** [comment]
@pdf -- our query engine has undergone significant refactoring for the 0.9.2 release. Can you rebase this change?

**Comment by pdf** [comment]
I'm not sure this is actually useful due to the order that fills are applied.

**Comment by otoolep** [comment]
@pdf -- our code has changed significantly since this PR due to the new engines and performance improvements. This will need to be re-implemented if you want to get it in.

**Comment by pdf** [comment]
@otoolep consider this abandoned, unless I roll around to InfluxDB again.

**Pull request closed without merging**